src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs

```diff
@@ -21,31 +23,79 @@ namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
     /// there won't be multiple threads calling this client at the same time.
     /// </remarks>
     internal class ConfigurationClientManager : IConfigurationClientManager
-    {
-        private readonly IList<ConfigurationClientWrapper> _clients;
-
-        public ConfigurationClientManager(IEnumerable<string> connectionStrings, ConfigurationClientOptions clientOptions)
-        {
-            if (connectionStrings == null || !connectionStrings.Any())
-            {
-                throw new ArgumentNullException(nameof(connectionStrings));
-            }
-
-            _clients = connectionStrings.Select(connectionString => 
-            {
-                var endpoint = new Uri(ConnectionStringParser.Parse(connectionString, ConnectionStringParser.EndpointSection));
-                return new ConfigurationClientWrapper(endpoint, new ConfigurationClient(connectionString, clientOptions));
-            }).ToList();
-        }
-
-        public ConfigurationClientManager(IEnumerable<Uri> endpoints, TokenCredential credential, ConfigurationClientOptions clientOptions)
-        {
-            if (endpoints == null || !endpoints.Any())
-            {
-                throw new ArgumentNullException(nameof(endpoints));
-            }
-
-            _clients = endpoints.Select(endpoint => new ConfigurationClientWrapper(endpoint, new ConfigurationClient(endpoint, credential, clientOptions))).ToList();
+    {
+        private IList<ConfigurationClientWrapper> _clients;
+
+        private IEnumerable<string> _connectionStrings;
+        private IEnumerable<Uri> _endpoints;
+        private TokenCredential _credential;
+        private ConfigurationClientOptions _clientOptions;
+        private Lazy<SrvLookupClient> _srvLookupClient;
+        private DateTimeOffset _lastFallbackClientRefresh = default;
+        private bool _allClientFailed = false;

```

Comment:
what is this for?

Reply:
For indicating whether it's the first attempt to get available clients when all clients failed. If it's first attempt, should perform dynamic client immediately, if not, it can wait till the _fallbackClientRefreshInterval due